### PR TITLE
ci(docker): dynamically generate the matrix

### PR DESCRIPTION
**Pull request by AtomicFS**
- instead of hard-coded matrix of docker containers, generate it automatically
- I also include matrix subsets only with `coreboot` and `edk2` entries, they will be needed later

### CODE REVIEW

**Comment by AtomicFS** [comment]
@MDr164 review pls?

File: .github/workflows/docker-build-and-test.yml [hunk 1]
```diff
@@ -30,36 +30,61 @@ permissions:
...
+  get-matrix:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: get-matrix
+        id: get-matrix
+        run: |
+          # Disable SC2046: Quote this to prevent word splitting
+          # I can't quite this, just look at it
```
Comment (MDr164) [inline .github/workflows/docker-build-and-test.yml:46]: I think there is either a word missing here or the `e` is too much in quite

**Review by MDr164 — COMMENTED** [review]
Looks good so far, just a minor error in a comment

File: .github/workflows/docker-build-and-test.yml (hunk 1 above)
Comment (AtomicFS) [reply .github/workflows/docker-build-and-test.yml]: Typo :)

**Pull request merged**